src/test/java/uk/gov/homeoffice/digital/sas/balancecalculator/BalanceCalculatorTest.java

```diff
@@ -0,0 +1,229 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.homeoffice.digital.sas.balancecalculator.client.RestClient;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Accrual;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.Agreement;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.accrual.enums.AccrualType;
+import uk.gov.homeoffice.digital.sas.balancecalculator.models.timecard.TimeEntry;
+import uk.gov.homeoffice.digital.sas.balancecalculator.utils.TestUtils;
+
+@ExtendWith(MockitoExtension.class)
+class BalanceCalculatorTest {
+
+  public static final AccrualType ANNUAL_TARGET_HOURS_TYPE =
+      AccrualType.ANNUAL_TARGET_HOURS;
+  public static final String ANNUAL_TARGET_HOURS_TYPE_ID =
+      ANNUAL_TARGET_HOURS_TYPE.getId().toString();
+  private static final ZonedDateTime SHIFT_START_TIME =
+      ZonedDateTime.parse("2023-04-18T08:00:00+00:00");
+  private static final ZonedDateTime SHIFT_END_TIME =
+      ZonedDateTime.parse("2023-04-18T10:00:00+00:00");
+  private static final LocalDate ACCRUAL_DATE = SHIFT_START_TIME.toLocalDate();
+  private static final BigDecimal SHIFT_DURATION = new BigDecimal(120);
+  private static final Range<ZonedDateTime> DATE_TIME_RANGE =
+      Range.closed(SHIFT_START_TIME, SHIFT_END_TIME);
+  private static final String TIME_ENTRY_ID = "7f000001-879e-1b02-8187-9ef1640f0003";
+  private static final String TENANT_ID = "52a8188b-d41e-6768-19e9-09938016342f";
+  private static final String PERSON_ID = "0936e7a6-2b2e-1696-2546-5dd25dcae6a0";
+  private static final String AGREEMENT_ID = "c0a80193-87a3-1ff0-8187-a3bfe2b80004";
+  private static final LocalDate AGREEMENT_END_DATE = LocalDate.of(2024, 3, 31);
+  @Mock
+  private RestClient restClient;
+
+  private BalanceCalculator balanceCalculator;
+
+  private static Stream<Arguments> testData() {
+    return Stream.of(
+        Arguments.of(TIME_ENTRY_ID, BigDecimal.valueOf(6600), BigDecimal.valueOf(7200),
+            BigDecimal.valueOf(7440), BigDecimal.valueOf(8160)),
+        Arguments.of("e7d85e42-f0fb-4e2a-8211-874e27d1e888", BigDecimal.valueOf(6240),
+            BigDecimal.valueOf(6840), BigDecimal.valueOf(7080), BigDecimal.valueOf(7800))
+    );
+  }
+
+  @BeforeEach
+  void setup() {
+    balanceCalculator = new BalanceCalculator(restClient);
+  }
+
+  @ParameterizedTest
+  @MethodSource("testData")
+  void calculateAccruals_withinCalendarDayAndAnnualTargetHours_returnUpdateAccrual(String timeEntryId,
+      BigDecimal expectedCumulativeTotal1, BigDecimal expectedCumulativeTotal2,
+      BigDecimal expectedCumulativeTotal3, BigDecimal expectedCumulativeTotal4)
+      throws IOException {
+    ZonedDateTime shiftStartTime = ZonedDateTime.parse("2023-04-18T08:00:00+00:00");
```

Comment:
I think the start and end dates should be parameterised in the test otherwise the test is too specific.
Also add a test to cover BST.

Reply:
Made start and end times into parameters and included a Daylight saving scenario